packages/integration-sdk-runtime/src/execution/__tests__/utils/getIngestionSourceStepStartStates.test.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  Step,
+  StepStartStates,
+  DisabledStepReason,
+  StepExecutionContext,
+} from '@jupiterone/integration-sdk-core';
+import { getIngestionSourceStepStartStates } from '../../utils/getIngestionSourceStepStartStates';
+
+const createStep = ({
+  id,
+  ingestionSourceId,
+}: {
+  id: string;
+  ingestionSourceId?: string;
+}) => ({
+  id,
+  ingestionSourceId,
+  name: id,
+  entities: [],
+  relationships: [],
+  executionHandler: jest.fn(),
+});
+
+describe('getIngestionSourceStepStartStates', () => {
+  const mockIntegrationSteps: Step<StepExecutionContext>[] = [
+    createStep({
+      id: 'step-1',
+      ingestionSourceId: 'source-1',
+    }),
+    createStep({
+      id: 'step-2',
+      ingestionSourceId: 'source-2',
+    }),
+    createStep({
+      id: 'step-3',
+    }),
+  ];
+
+  const mockConfigStepStartStates: StepStartStates = {
+    'step-1': { disabled: true, disabledReason: DisabledStepReason.PERMISSION },
+    'step-2': { disabled: false },
+    'step-3': { disabled: false },
+  };
+
+  it('keeps a disabled step as disabled with the same reason', () => {
```

Comment:
This text is a bit confused, I think you are trying to say that it keeps the step configuration from the `getStepStartStates` from the integration, so what do you think of something like:
```
prioritizes the step state from the instance's getStepStartStates
```
Or something like that...